StRoot/StFwdUtils/StFwdAnalysisMaker.h

```diff
@@ -27,6 +18,51 @@ class StFwdAnalysisMaker : public StMaker {
     void Clear(const Option_t *opts = "");
     void ProcessFwdTracks();
     void ProcessFwdMuTracks();
+
+    // StEvent analyzed by default
+    // call this to analyze the MuDst instead
+    void setMuDstInput() { mAnalyzeMuDst = true; }
+
+  protected:
+
+    /**
+     * @brief Map of <name (TString), histogram>
+     * 
+     */
+    std::map<TString, TH1*> mHists;
+
+    /**
+     * @brief Add a histogram to the map
+     * 
+     * Convenience method to avoid duplicate typing name / possible mismatch
+     * @param h1 TH1* histogram object
+     * @return TH1* return the same object for ease of use
+     */
+    TH1 * addHist( TH1 * h1){
+      mHists[h1->GetName()] = h1;
+      return h1;
+    }
+
+    /**
+     * @brief Get the Hist object from the map
+     *  - Additional check and safety for missing histograms
+     * @param n Histogram name
+     * @return TH1* histogram if found, otherwise a 'nil' histogram with one bin
+     */
+    TH1* getHist( TString n ){
+      if (mHists.count(n))
+        return mHists[n];
+      LOG_ERROR << "Attempting to access non-existing histogram" << endm;
+      return new TH1F( "NULL", "NULL", 1, 0, 1 ); // returning nullptr can lead to seg fault, this fails softly
```

Comment:
Why not return 0?

Reply:
I guess you want the code to run without crashing even if a histogram isn't present. But I also see that the expected histogram is sometimes a TH2F, not a TH1F, on which you call `Fill()` with two doubles as arguments....and luckily this is defined for a TH1F, but if you ever add a TH3F you will have an error.

Reply:
I left this as is, indeed it will be an issue with TH3, but I do not plan to use any. But will keep in mind if that changes.